module/executiondatasync/pruner/pruner.go

```diff
@@ -0,0 +1,206 @@
+package pruner
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/module"
+	"github.com/onflow/flow-go/module/component"
+	"github.com/onflow/flow-go/module/executiondatasync/tracker"
+	"github.com/onflow/flow-go/module/irrecoverable"
+	"github.com/onflow/flow-go/module/util"
+)
+
+const (
+	defaultHeightRangeTarget = uint64(400_000)
+	defaultThreshold         = uint64(100_000)
+)
+
+// Pruner is a component responsible for pruning data from
+// execution data storage. It is configured with the following
+// parameters:
+// * Height range target: The target number of most recent blocks
+//   to store data for. This controls the total amount of data
+//   stored on disk.
+// * Threshold: The number of block heights that we can exceed
+//   the height range target by before pruning is triggered. This
+//   controls the frequency of pruning.
```

Comment:
```suggestion
//   reduce the frequency of pruning, since pruning is implemented by batch removal,
//   and is more efficient when pruning a big range of height at once.
```